frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -758,7 +758,7 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
         this.conditions.controls.forEach((control, index) => {
           const assignmentWeightFormControl = control.get(SIMPLE_EXP_CONSTANTS.FORM_CONTROL_NAMES.ASSIGNMENT_WEIGHT);
           assignmentWeightFormControl.setValue(
-            control.value.assignmentWeight ? control.value.assignmentWeight : this.previousAssignmentWeightValues[index]
+            control.value.assignmentWeight ? control.value.assignmentWeight : 0
```

Comment:
are there are no scenarios here where we would want the previous assignment weight instead of 0 when assignment weight is undefined?

Reply:
the previous assignment weight assigned value only if equal weight is true and I didn't change it for equal weight case, but if equal weight isn't true then it is just empty array replacing our 0 value to undefined when we try to read any index value. I think even if we remove that line it will be fine but if in any case we get null or undefined weight this will also replace it with 0 and not throw error so it's win win in my view.